hack/codesign.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+set -eu
+
+# Code Signing
+# ============
+# This script facilitates code signing of binaries for macOS. This script conditionally signs a binary
+# under the following conditions:
+# 1. The binary path contains "darwin" (this prevents duplicating GoReleaser build configurations)
+# 2. The `security` and `codesign` programs are available (this allows the script to be run on Linux)
+# 3. The P12 key pair used as the signing identity is available (allows the script to be run on non-release builds)
+
+# Verify, but exit normally to ensure the conditional nature of the checks
+FILE="${1:?missing file argument}"
+[ "$(basename "$(dirname "$FILE")")" == "redskyctl_darwin_amd64" ] || { echo >&2 "skipping code signing on file '$1'"; exit; }
+command -v security >/dev/null 2>&1 || { echo >&2 "skipping code signing, security not present"; exit; }
```

Comment:
Should we flex based on `uname -s` instead?

```
if [ $(uname -s) != "Darwin" ]; then exit; fi
```

Reply:
I figured it was better to test that the tools were there, e.g. a Mac without Xcode installed.